tests/unit/spec/base/stub-attribution.js

```diff
@@ -536,20 +554,66 @@ describe('stub-attribution.js', function () {
         });
     });
 
-    describe('waitForGoogleAnalytics', function () {
-        beforeEach(function () {
-            // stub out GA client ID
-            spyOn(Mozilla.StubAttribution, 'getGAClientID').and.returnValue(
+    describe('waitForGoogleAnalyticsThen', function () {
+        it('should fire a callback when GA client ID and session ID are found', function () {
+            // stub out UA client ID
+            spyOn(Mozilla.StubAttribution, 'getUAClientID').and.returnValue(
                 GA_CLIENT_ID
             );
+            // stub out GA4 client ID
+            spyOn(Mozilla.StubAttribution, 'getGtagClientID').and.returnValue(
+                GA4_CLIENT_ID
+            );
+            const callback = jasmine.createSpy('callback');
+
+            Mozilla.StubAttribution.waitForGoogleAnalyticsThen(callback);
+            expect(callback).toHaveBeenCalledWith(true);
         });
 
-        it('should fire a callback when GA client ID and session ID are found', function () {
+        it('should fire a callback when UA client ID is found but GA4 client ID is not', function () {
+            // stub out UA client ID
+            spyOn(Mozilla.StubAttribution, 'getUAClientID').and.returnValue(
+                GA_CLIENT_ID
+            );
+            // stub out GA4 client ID
+            spyOn(Mozilla.StubAttribution, 'getGtagClientID').and.returnValue(
+                null
+            );
             const callback = jasmine.createSpy('callback');
 
-            Mozilla.StubAttribution.waitForGoogleAnalytics(callback);
+            Mozilla.StubAttribution.waitForGoogleAnalyticsThen(callback);
```

Comment:
I think this code is failing because jasmine's own internal clock is not waiting for the code's `setTimeout()` to run its course. To fix this you can use [clock](https://jasmine.github.io/api/5.1/Clock.html) and then tick the clock forward, forcing the timeouts to execute. Something like:

```javascript
beforeEach(function () {
    jasmine.clock().install();
});

afterEach(function () {
    jasmine.clock().uninstall();
});

it('should fire a callback when UA client ID is found but GA4 client ID is not', async function () {
    // stub out UA client ID
    spyOn(Mozilla.StubAttribution, 'getUAClientID').and.returnValue(
        GA_CLIENT_ID
    );
    // stub out GA4 client ID
    spyOn(Mozilla.StubAttribution, 'getGtagClientID').and.returnValue(
        null
    );
    const callback = jasmine.createSpy('callback');

    const result = await Mozilla.StubAttribution.waitForGoogleAnalyticsThen(callback);
    jasmine.clock().tick(2100);
    expect(callback).toHaveBeenCalledWith(true);
});
```

The nice thing here is it doesn't actually slow down the tests :)

Reply:
Fixed it, thanks!